### PR TITLE
system tests: new random_free_port helper

### PR DESCRIPTION
**Pull request by edsantiago**
Picks a pseudorandom open port within a range. Refactor existing
instances of such code.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

File: test/system/500-networking.bats [hunk 1]
```diff
@@ -188,6 +185,7 @@ load helpers
 
 # "network create" now works rootless, with the help of a special container
 @test "podman network create" {
+    # Deliberately use a fixed port, not random_open_port, because of #10806
     myport=54322
```
Comment (Luap99) [inline test/system/500-networking.bats:189]: line 245 also uses a fixed port, can this be changed?

**Review by Luap99 — APPROVED** [review]
LGTM, one question.

**Comment by mheon** [comment]
LGTM

**Pull request merged**